runner/cf-executor/cg/creds_test.go

```diff
@@ -0,0 +1,100 @@
+package cg
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+var syntaxError *json.SyntaxError
+
+func getVcapJson(u string, p string) string {
+	return fmt.Sprintf(`{"cloud-gov-service-account":[{"credentials":{"username":"%s","password":"%s"}}]}`, u, p)
+}
+
+func Test_getCreds(t *testing.T) {
+	tests := []struct {
```

Comment:
[praise] I really love this test setup, super clear about what we're testing for and what we're expecting.

Reply:
…_and_ super easy to add new test cases! I like it too. I hadn't encountered it as a commonly used pattern before but the Go community calls it _table-driven testing_, and it seems to be more generically referred to as _data-driven testing_… which is _maybe_ more commonly used, as a named pattern at least, by Java folk and people doing automated testing of complex systems.

It's common enough in Go-land that it seems to be almost a _default_. There is a nice little CLI tool called [`gotests`](https://github.com/cweill/gotests) (with several editor plugins) that generates handy table testing stubs like this.